src/main/java/calculator/Calculator.java

```diff
@@ -0,0 +1,54 @@
+package calculator;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class Calculator {
+
+    private final List<Integer> numberQueue = new ArrayList<>();
+    private final List<Operator> operatorQueue = new ArrayList<>();
+
+    public void updateExpression(String expressionString) {
+        List<String> elements = Arrays.asList(expressionString.split(" "));
+        try {
+            numberQueue.add(Integer.valueOf((elements.get(0))));
+
+            for (int i = 1; i < elements.size(); i = i + 2) {
+                operatorQueue.add(Operator.find(elements.get(i)));
+                numberQueue.add(Integer.valueOf((elements.get(i + 1))));
+            }
+        } catch (NumberFormatException | ArrayIndexOutOfBoundsException ex) {
+            throw new IllegalArgumentException("계산할 수 없는 계산식 입니다.");
+        }
+
+        if (!validation()) {
+            throw new IllegalArgumentException("계산할 수 없는 계산식 입니다.");
+        }
+    }
+
+    public int calculate() {
+        int result = numberQueue.get(0);
+        for (int operatorIndex = 0, numberIndex = 1;
+            operatorIndex < operatorQueue.size();
+            operatorIndex++, numberIndex++) {
+            Operator operator = operatorQueue.get(operatorIndex);
+            int number = numberQueue.get(numberIndex);
+            result = operator.calculate(result, number);
+        }
+
+        return result;
+    }
+
+    private boolean validation() {
+        if (numberQueue.size() < 1) {
+            return false;
+        }
+
+        if (numberQueue.size() != operatorQueue.size() + 1) {
+            return false;
+        }
+
+        return true;
+    }
```

Comment:
이 부분은 한줄로 작성이 가능하겠네요 :) 

```suggestion
        return numberQueue.size() == operatorQueue.size() + 1;
    }
```

Reply:
만약 validation() 안에 if 문이 한개였다면 위 방식을 사용했겠지만, 이미 2개의 if 문이 존재하기에 가독성+일관성을 위해 if 를 분리했습니다.